File: src/main/java/me/kuku/yuq/entity/QQLoginEntity.java
package me.kuku.yuq.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import me.kuku.utils.QqUtils;

import javax.persistence.*;

@Entity
@Table(name = "qqLogin")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class QQLoginEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column(unique = true)
    private Long qq;
    @Column(name = "group_")
    private Long group;
    private String password;
    private String sKey;
    private String psKey;
    private String groupPsKey;
    private String superKey;
    private String superToken;
    private String pt4Token;
    private Boolean status;

    public String getCookie(){
        return String.format("pt2gguin=o0%s; uin=o0%s; skey=%s; ", qq, qq, sKey);
    }

    public String getCookie(String psKey){
        return String.format("%sp_skey=%s; p_uin=o0%s;", getCookie(), psKey, qq);
    }

    public String getCookieWithQQZone(){
        return String.format("%sp_skey=%s; p_uin=o0%s; ", getCookie(), psKey, qq);
    }

    public String getCookieWithGroup(){
        return String.format("%sp_skey=%s; p_uin=o0%s; ", getCookie(), groupPsKey, qq);
    }

    public String getCookieWithSuper(){
        return String.format("superuin=o0%s; superkey=%s; supertoken=%s; ", qq, superKey, superToken);
    }

    public String getGtk(){
        return String.valueOf(QqUtils.getGTK(sKey));
    }

    public String getGtk(String psKey){
        return String.valueOf(QqUtils.getGTK(psKey));
    }

    public String getGtk2(){
        return QqUtils.getGTK2(sKey);
    }

    public String getGtkP(){
        return String.valueOf(QqUtils.getGTK(psKey));
    }

    public String getToken(){
        return String.valueOf(QqUtils.getToken(superToken));
    }

}

File: src/main/kotlin/me/kuku/yuq/entity/QQLoginEntity.kt
package me.kuku.yuq.entity

import me.kuku.yuq.utils.OkHttpClientUtils
import me.kuku.yuq.utils.QQUtils
import javax.persistence.*

@Entity
@Table(name = "qqLogin")
data class QQLoginEntity(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Int? = null,
        @Column(unique = true)
        var qq: Long = 0L,
        var qqGroup: Long = 0L,
        var password: String = "",
        var sKey: String = "",
        var psKey: String = "",
        var groupPsKey: String? = "",
        var superKey: String = "",
        var superToken: String = "",
        var pt4Token: String = "",
        var status: Boolean = false
)
{
    @Transient
    fun cookie() = OkHttpClientUtils.addCookie(this.getCookie())
    @Transient
    fun cookieWithQQZone() = OkHttpClientUtils.addCookie(this.getCookieWithQQZone())
    @Transient
    fun cookieWithGroup() = OkHttpClientUtils.addCookie(this.getCookieWithGroup())
    @Transient
    fun getCookie() = "pt2gguin=o0$qq; uin=o0$qq; skey=$sKey; "
    @Transient
    fun getCookie(psKey: String) = "${this.getCookie()}p_skey=$psKey; p_uin=o0$qq;"
    @Transient
    fun getCookieWithQQZone() = "${this.getCookie()}p_skey=$psKey; p_uin=o0$qq; "
    @Transient
    fun getCookieWithGroup() = "${this.getCookie()}p_skey=$groupPsKey; p_uin=o0$qq; "
    @Transient
    fun getCookieWithSuper() = "superuin=o0$qq; superkey=$superKey; supertoken=$superToken; "
    @Transient
    fun getGtk() = QQUtils.getGtk(sKey).toString()
    @Transient
    fun getGtk(psKey: String) = QQUtils.getGtk(psKey).toString()
    @Transient
    fun getGtk2() = QQUtils.getGtk2(sKey)
    @Transient
    fun getGtkP() = QQUtils.getGtk(psKey).toString()
    @Transient
    fun getToken() = QQUtils.getToken(superToken).toString()

}